src/plugins/auto/auto_schedule.cpp

```diff
@@ -125,105 +124,181 @@ void AutoSchedule::init(const ScheduleContext::Ptr& sContext) {
+            // need lock
+            {
+                std::lock_guard<std::mutex> lock(_autoSContext->_confMutex);
+                _autoSContext->_config.insert(contextPtr->deviceInfo.config.begin(),
+                                              contextPtr->deviceInfo.config.end());
+            }
+            contextPtr->isAlready = true;
+            auto& deviceName = contextPtr->deviceInfo.deviceName;
+            LOG_INFO_TAG("device:%s loading Network finished", deviceName.c_str());
+            if (isCumulative) {
+                auto optimalNums = contextPtr->executableNetwork->GetMetric(ov::optimal_number_of_infer_requests.name())
+                                       .as<unsigned int>();
+                LOG_INFO_TAG("device:%s, optimal_infer_number:%d", deviceName.c_str(), optimalNums);
+                std::lock_guard<std::mutex> lock(_autoSContext->_confMutex);
+                _autoSContext->_ctputOtimalNums += optimalNums;
+            }
+            auto supported_config_keys = _autoSContext->_core->GetMetric(deviceName, METRIC_KEY(SUPPORTED_CONFIG_KEYS))
+                                             .as<std::vector<std::string>>();
+            DEBUG_RUN([this, &contextPtr, &deviceName, &supported_config_keys] {
+                std::lock_guard<std::mutex> lock(_autoSContext->_confMutex);
+                for (const auto& cfg : supported_config_keys) {
+                    try {
+                        LOG_DEBUG_TAG("device:%s, GetConfig:%s=%s",
+                                      deviceName.c_str(),
+                                      cfg.c_str(),
+                                      contextPtr->executableNetwork->GetConfig(cfg).as<std::string>().c_str());
+                    } catch (const IE::Exception&) {
+                    }
+                }
+            });
+        }
+        // Handle device load failure in case of ctput
+        if (isCumulative && !contextPtr->isLoadSuccess) {
+            std::string failedDeviceName = contextPtr->deviceInfo.deviceName;
+            std::lock_guard<std::mutex> lock(_autoSContext->_confMutex);
+            const auto DeviceIter =
+                std::find_if(_autoSContext->_devicePriorities.begin(),
+                             _autoSContext->_devicePriorities.end(),
+                             [&](const DeviceInformation& d) -> bool {
+                                 return d.deviceName.find(failedDeviceName) != std::string::npos;
+                             });
+            // Remove failed device from _devicePriorities
+            if (DeviceIter != _autoSContext->_devicePriorities.end()) {
+                _autoSContext->_devicePriorities.erase(DeviceIter);
+            }
+            // Remove failed device from ov::device::priorities in config
+            auto it_prior = _autoSContext->_config.find(ov::device::priorities.name());
+            if (it_prior != _autoSContext->_config.end()) {
```

Comment:
why do we have to remove the config here, this is set by user right?

Reply:
AutoExecutableNetwork::GetMetric(ov::device::priorities.name())  does not return the  fail loaded device, so remove failed  device from ov::device::priorities.